### PR TITLE
Improve test coverage in `internal/vsrpc`

### PR DESCRIPTION
**Pull request by ellismg**
This change adds coverage for the `GetAspireHostAsync` RPC in `AspireService` as well as some of the session logic that is common across all services.

While writing the tests, I realized that it would be both easier and more correct to use `project.Load` directly in `GetAspireHostAsync` instead of trying to resolve a `*ProjectConfig` from the IoC container. This better reflects the intention of the code. In practice, the production implementation of the resolve function just called `project.Load` anyway, so there is no functional difference. However, in cases where the `azure.yaml` is corrupt in some manner, we'll have better error messages here as well.

### CODE REVIEW

File: cli/azd/internal/vsrpc/util_test.go
Comment (ellismg) [inline cli/azd/internal/vsrpc/util_test.go:17]: This stuff is starting to get splayed across the tree (for both our acceptance tests, where we actually run the `dotnet` toolchain against them) and the unit level tests in aspire (where it's helpful to have these artifacts on disk even without running them (and instead mock the behavior of .NET).

I want to centralize all this stuff in a `internal/samples` package (that has all the embeds exported as well as `coptSample` and friends and sort of make this common technology our tests can leverage.

Can do that before landing this or after depending on how everyone feels - I wanted to just unblock myself on the plane and focus on writing these tests vs a larger samples refactoring.

Interested in your thoughts here, @weikanglim, since you had been writing tests that need similar technolog?  I sort of find value in having a few different shapes of aspire apps and manifests for my testing.

File: cli/azd/internal/vsrpc/testdata/manifests/aspire-starter.json
Comment (ellismg) [inline cli/azd/internal/vsrpc/testdata/manifests/aspire-starter.json:1]: I will add a script to regenerate this based on the sample in `samples/aspire-starter.json`

File: cli/azd/internal/vsrpc/util_test.go
Comment (weikanglim) [reply cli/azd/internal/vsrpc/util_test.go:17]: I haven't given this thought the amount of duration it deserves, but I really like the locality between the fixtures for the test and the test itself is nice, so my initial inkling is to lean away from having a central location.

But having a library that simplifies some of this work does make a lot of sense. I draw heavy inspiration from some projects I see out there, where there's an "acceptance test library" that is heavily used to avoid the extra boilerplate, but each go package and test package is largely independent from one another.

**Review by vhvb1989 — APPROVED** [review]
LGTM

**Pull request closed without merging**
